aws-csharp/${PROJECT}.csproj

```diff
@@ -4,6 +4,7 @@
     <OutputType>Exe</OutputType>
     <TargetFramework>net8.0</TargetFramework>
     <Nullable>enable</Nullable>
+    <DefaultItemExcludes>$(DefaultItemExcludes);tests/**/*.cs;sdks/**/*.cs</DefaultItemExcludes>
```

Comment:
The more idiomatic thing to do would be to have the program and test projects SxS:

```
proj/
├── Pulumi.yaml
├── Program.cs
└── proj.csproj
tests/
├── InfraTests.cs
└── Tests.csproj
```

But we can't easily do that from a template. If we structured the template like this, you wouldn't be able to run `pulumi up` after `pulumi new` because the program would live in a subdirectory. You'd have to `cd` into the program's subdirectory first.

I think we want to keep it such that you can run `pulumi up` immediately after `pulumi new`, without having to change directories. So we nest the test project:

```
proj/
├── Pulumi.yaml
├── Program.cs
├── proj.csproj
└── tests/
    ├── InfraTests.cs
    └── Tests.csproj
```

To do so, we have to exclude the source files in the nested project here.

We have a similar issue with parameterized providers, where `pulumi package add` generates the sdks in a `sdks` subdirectory, and we ask users to add `<DefaultItemExcludes>` for that. For that reason, I've included excludes for both `sdks/` and `tests/` by default.